src/dehydrated_devices.rs

```diff
@@ -33,18 +78,54 @@ impl DehydratedDevices {
     #[wasm_bindgen]
     pub async fn rehydrate(
         &self,
-        pickle_key: &Uint8Array,
+        pickle_key: &DehydratedDeviceKey,
         device_id: &DeviceId,
         device_data: &str,
     ) -> Result<RehydratedDevice, JsError> {
-        let pickle_key: [u8; 32] =
-            pickle_key.to_vec().try_into().map_err(|_| JsError::new("Wrong key length"))?;
+        let sdk_pickle_key =
+            SdkDehydratedDeviceKey::from_slice(pickle_key.inner.to_vec().as_slice())?;
+
         Ok(self
             .inner
-            .rehydrate(&pickle_key, &device_id.inner, serde_json::from_str(device_data)?)
+            .rehydrate(&sdk_pickle_key, &device_id.inner, serde_json::from_str(device_data)?)
             .await?
             .into())
     }
+
+    /// Get the cached dehydrated device pickle key if any.
+    ///
+    /// None if the key was not previously cached (via
+    /// [`Self::save_dehydrated_device_pickle_key`]).
```

Comment:
is it correct that `save_dehydrated_device_pickle_key` is the only way to save the key? It is not loaded automatically (eg via secret sharing) at any point?

Reply:
this part is not yet done.

Reply:
Ok. We'll need to remember to update this comment when this changes.